satellite/gracefulexit/endpoint.go

```diff
@@ -620,3 +625,39 @@ func (endpoint *Endpoint) handleFailed(ctx context.Context, pending *pendingMap,
 
 	return nil
 }
+
+func (endpoint *Endpoint) updatePointer(ctx context.Context, exitingNodeID storj.NodeID, receivingNodeID storj.NodeID, path []byte, pieceNum int32) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	// remove the node from the pointer
+	pointer, err := endpoint.metainfo.Get(ctx, string(path))
+	if err != nil {
+		return Error.Wrap(err)
+	}
+	remote := pointer.GetRemote()
+	// nothing to do here
+	if remote == nil {
+		return nil
+	}
+
+	var toRemove []*pb.RemotePiece
+	for _, piece := range remote.GetRemotePieces() {
+		if piece.NodeId == exitingNodeID {
+			toRemove = []*pb.RemotePiece{piece}
+			break
+		}
+	}
+	var toAdd []*pb.RemotePiece
+	if !receivingNodeID.IsZero() {
```

Comment:
Is the excluded `else` intentional? If we use this function for deleting pieces from the pointer we probably don't want to return an error if there is no receiving node ID, so this is probably fine. I just wanted to make sure that was your intention.

Reply:
It was intentional